nova/virt/vmwareapi/ds_util.py

```diff
@@ -443,28 +434,28 @@ def get_sub_folders(session, ds_browser, ds_path):
     return set()
 
 
-def _filter_datastores_matching_storage_policy(session, data_stores,
+def _filter_datastores_matching_storage_policy(session, datastores,
                                                storage_policy):
     """Get datastores matching the given storage policy.
 
-    :param data_stores: the list of retrieve result wrapped datastore objects
+    :param datastores: the list of datastore results
     :param storage_policy: the storage policy name
     :return: the list of datastores conforming to the given storage policy
     """
     profile_id = pbm.get_profile_id_by_name(session, storage_policy)
+    datastores = list(datastores)
```

Comment:
The parameter is documented as list of datastores, but we copy the list here ... because it's really an iterator I guess and we iterate it twice?
And we now also don't return a list anymore (as documented) but instead this function is an iterator now?

Reply:
I've updated the documentation.

Reply:
I think we want to add a little `# NOTE: ` here then.

I don't see it passing the datastores to the PBM API - only the `ds_mors`, which is already a list.

Reply:
Well, it is a list after iterating over all `datastores`.
And then we have to do it again, to map it back to the original retrieval objects

Reply:
Looks dangerously like quadratic runtime

Reply:
I've replaced it with another explicit implementation, which should have a better runtime.